### PR TITLE
Earthstar 6 Spec

### PR DESCRIPTION
**Pull request by AljoschaMeyer**
Don't merge yet, I just want the preview deploy.

### CODE REVIEW

File: willowtest/specs/earthstar/es6_spec.ts [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+		},
+		[
+			table_of_contents(7),
+
+			pinformative(
+				"This document specifies version 6 of the ", link("Earthstar protocol", "https://earthstar-project.org/"), ". The protocol behind Earthstar is an instantiation of ", link_name("willow", "Willow"), ": the Earthstar data model is a particular instantiation of the ", link_name("data_model", "Willow data model"), ", using an instantiation of ", link_name("meadowcap", "Meadowcap"), " for access control, and synchronising data with the ", link_name("sync", "WGPS"), "."
+			),
+
+			pinformative(
+				"In addition to fixing a specific set of protocol parameters for Willow, Earthstar restricts the set of bytes that may appear in ", rs("Path"), " to the ascii encodings of the following characters: ", code("-.0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_"), ", that is, alphanumerics and ", code("-"), ", ", code("."), ", and ", code("_"), "."
```
Comment (sgwilym) [inline willowtest/specs/earthstar/es6_spec.ts:26]: I think this has been accidentally left in, unless you've had a change of heart?

File: willowtest/specs/earthstar/es6_spec.ts [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+				pinformative(
+					"The type ", r("NamespaceId"), " is the type of ", rs("es6_namespace"), "."
+				),
+
+				pinformative(
+					"The type ", r("SubspaceId"), " is the type of ", rs("es6_identity"), "."
+				),
+
+				pinformative(
+					"The ", r("max_component_length"), " is 64, the ", r("max_component_count"), " is 16, and the ", r("max_path_length"), " is 1024. Remember that Earthstar only allows bytes that are ascii encodings of the following characters in paths: ", code("-.0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz_"), "."
```
Comment (sgwilym) [inline willowtest/specs/earthstar/es6_spec.ts:89]: Likewise for this.

**Review by sgwilym — COMMENTED** [review]
Do you think it makes sense for me to add addendum on:

- Tag encoding (e.g. @gwil.bxxx...)?
- The convention of encoding 'friendl'y paths when it satisfies the currently defined charset?

File: willowtest/specs/earthstar/es6_spec.ts (hunk 1 above)
Comment (AljoschaMeyer) [reply willowtest/specs/earthstar/es6_spec.ts:26]: Good catch, this should go.

**Comment by AljoschaMeyer** [comment]
> Do you think it makes sense for me to add addendum

Sure, can't really hurt.

**Pull request merged**